module/VuFind/src/VuFind/View/Helper/Root/SearchMemory.php

```diff
@@ -73,14 +73,26 @@ public function __construct(Memory $memory)
      * @return string
      */
     public function getLastSearchLink($link, $prefix = '', $suffix = '')
+    {
+        if ($url = $this->getLastSearchUrl()) {
+            return $prefix . '<a href="' . $url . '">' . $link . '</a>' . $suffix;
+        }
+        return '';
+    }
+
+    /**
+     * If a previous search is recorded in the session, return its URL
```

Comment:
This method is currently returning an HTML-escaped version of the URL. Is that desirable? Would it be better to move the escaping to the calling code? If it makes sense to keep it here, we should probably rename the method and/or adjust the comment to reflect the escaping so people aren't confused/surprised.

Reply:
At some point I tried using this function to implement the feature, way I abandoned however I kept the refactorisation.
An I kept the function behaviour without altering it.
In the last commit, I moved the escaper in the function above (with the original name) instead of this one. What do you think of it that way?

Reply:
#3845